src/DirectoryCleaner.php

```diff
@@ -58,7 +69,13 @@ public function deleteFilesOlderThanMinutes($amountOfFilesDeleted = 0, $director
                 return $this->policy()->shouldDelete($file);
             })
             ->each(function ($file) {
-                $this->filesystem->delete($file);
+                if ($this->extensions){
+                    if (in_array($file->getExtension(),$this->extensions)){
+                        $this->filesystem->delete($file);
+                    }
+                }else{
```

Comment:
Avoid using `else`. Maybe you could add a protected function `shouldDelete` and use early returns there.